### ISSUE TITLE
configuring read only connection still uses default connection

### ISSUE THREAD

**Issue by pablobuenaposada** [groveco/django-sql-explorer#453]
Hi, I am using sql-explorer in a project, at first decided to use the default django connection and everything worked as expected, later on I decided to create a new database user with read only permissions for sql-explorer as your documentation suggests.
Unfortunately, after this change each of the queries I had in there was throwing me a 500 error, in detail, it was throwing:

> InvalidExplorerConnectionException: Attempted to access connection default, but that is not a registered Explorer connection.

After getting really mad and check the code I found what was the problem, the connection used for every query is somehow stored in the Query django model you guys have (which I don't know why) which of course it was still pointing to the default connection and not the new one, had to manually update all the `connection` fields of this Query model to point to the new connection like this:

> Query.objects.all().update(connection="sql_explorer")

after this everything is working fine again with the new connection, not sure why you guys want to store the connection but I think we are missing a mechanism to update those if you change EXPLORER_CONNECTIONS in settings.py at least.

**Comment by marksweb** [comment]
@chrisclark I think this might need your input as I suspect this is something you'd decided on.

**Comment by lawson89** [comment]
@marksweb @pablobuenaposada you are correct the connection is stored with the query. Maybe this was the start of being able to support queries to multiple backend connections?

**Comment by RahulSwami01** [comment]
> Hi, I am using sql-explorer in a project, at first decided to use the default django connection and everything worked as expected, later on I decided to create a new database user with read only permissions for sql-explorer as your documentation suggests. Unfortunately, after this change each of the queries I had in there was throwing me a 500 error, in detail, it was throwing:
> 
> > InvalidExplorerConnectionException: Attempted to access connection default, but that is not a registered Explorer connection.
> 
> After getting really mad and check the code I found what was the problem, the connection used for every query is somehow stored in the Query django model you guys have (which I don't know why) which of course it was still pointing to the default connection and not the new one, had to manually update all the `connection` fields of this Query model to point to the new connection like this:
> 
> > Query.objects.all().update(connection="sql_explorer")
> 
> after this everything is working fine again with the new connection, not sure why you guys want to store the connection but I think we are missing a mechanism to update those if you change EXPLORER_CONNECTIONS in settings.py at least.

@pablobuenaposada  Do you know the file/folder name to update the settings? I am also facing the same issue.

**Comment by lawson89** [comment]
@RahulSwami01 it's actually in the explorer_query table- look at the connection column

**Comment by RahulSwami01** [comment]
Getting following error 

<html>
<body>
<h1 style="padding: 0px; margin: 0px; font-weight: normal; color: rgb(0, 0, 0); font-family: sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">InvalidExplorerConnectionException at /explorer/1/</h1><pre class="exception_value" style="padding: 0px; margin: 10px 0px; font-size: 1.5em; white-space: pre-wrap; word-break: break-word; font-family: sans-serif; color: rgb(87, 87, 87); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Attempted to access connection other, but that is not a registered Explorer connection.</pre>

Request Method: | GET
-- | --
http://127.0.0.1:8000/explorer/1/
4.1.7
InvalidExplorerConnectionException
Attempted to access connection other, but that is not a registered Explorer connection.
C:\Users\RASWAMI\AppData\Local\Programs\Python\Python310\lib\site-packages\explorer\utils.py, line 181, in get_valid_connection
explorer.views.query.QueryView
C:\Users\RASWAMI\AppData\Local\Programs\Python\Python310\python.exe
3.10.5
['C:\\name\\py\\oneclick',  'C:\\Users\\RASWAMI\\AppData\\Local\\Programs\\Python\\Python310\\python310.zip',  'C:\\Users\\RASWAMI\\AppData\\Local\\Programs\\Python\\Python310\\DLLs',  'C:\\Users\\RASWAMI\\AppData\\Local\\Programs\\Python\\Python310\\lib',  'C:\\Users\\RASWAMI\\AppData\\Local\\Programs\\Python\\Python310',  'C:\\Users\\RASWAMI\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages',  'C:\\Users\\RASWAMI\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\win32',  'C:\\Users\\RASWAMI\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\win32\\lib',  'C:\\Users\\RASWAMI\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\Pythonwin']
Wed, 15 Feb 2023 19:06:58 +0000


</body>
</html>

**Comment by lawson89** [comment]
@RahulSwami01  Did you also change EXPLORER_DEFAULT_CONNECTION in settings.py?

**Comment by RahulSwami01** [comment]
DATABASES = {
			'other': {
		        'ENGINE': 'django.db.backends.mysql',
		        'HOST': 'somehostname',
		        'PORT': '1433',
		        'NAME': 'DBName',
		        'USER': 'Username',
		        'PASSWORD': 'Password'
		       
		    },
                        'default': {
                        'ENGINE': 'django.db.backends.sqlite3',
                        'NAME': BASE_DIR / 'db.sqlite3'
                    },
		}

**Comment by lawson89** [comment]
Yeah I think you should find this setting (which is probably set to default and change it to other)
EXPLORER_DEFAULT_CONNECTION='other'

**Comment by RahulSwami01** [comment]
I have the following connection strings: and i am trying to connect to the new database "other" mentioned above. 
it has only read access. 

EXPLORER_CONNECTIONS = { 'Default': 'default'}
EXPLORER_DEFAULT_CONNECTION = 'default'

**Comment by RahulSwami01** [comment]
I made it, it shows me loading......loading...... and not showing anything.

EXPLORER_CONNECTIONS = { 'Default': 'default',
                         'other':'other'}
EXPLORER_DEFAULT_CONNECTION = 'other'

**Comment by RahulSwami01** [comment]
Following query I am using to test it: 
Select DATE('now') AS 'CurrentDATETime'

**Comment by lawson89** [comment]
Ok and you updated all the queries in the explorer_query table, connection column to point to 'other'?
I'd also make sure that you can connect using another tool to the mysql db using the credentials in your settings.py
One more thing - I'd restart the django test server as well (you probably already did this)

if this still doesn't work let me know and I'll test a similar setup this evening after work

**Comment by lawson89** [comment]
Oh - looks like you are using mysql django backend ('django.db.backends.mysql') but connecting to a microsoft sql server instance?

**Comment by RahulSwami01** [comment]
do you think that is the issue?

**Comment by lawson89** [comment]
I haven't ever used Django with MS SQL but yeah it requires a 3rd party lib to be installed and configured

https://docs.djangoproject.com/en/4.1/ref/databases/#third-party-notes

**Comment by lawson89** [comment]
@RahulSwami01 I think your best bet is to figure out how to get your Django app connected up to MS SQL server if that's the db you need to use. Then everything else will just work.

**Comment by RahulSwami01** [comment]
It’s been 4 days, I found out that it only works with the old version
(Django). For sql explorer we need new version (Django). So it’s deadlock
for me.

**Comment by lawson89** [comment]
@RahulSwami01 What version do you need to use? Looking at the tox seyup - looks like sql explorer is tested on 3.2 and above. It *may* work on earlier versions (@marksweb ?)

**Comment by marksweb** [comment]
Yeah version 3 dropped support for <3.2 of django.
